lib/omnibus/util.rb

```diff
@@ -65,18 +99,56 @@ def shellout(*args)
         options[:environment] = options.fetch(:environment, {}).merge(options[:env])
       end
 
+      # Double check that we don't have conflicting PATH definitions.
+      path_keys = options[:environment].keys.grep(/\Apath\Z/i)
+      if path_keys.length > 1
+        raise InvalidValue.new("shellout", "receive environment without duplicate PATH values")
+      elsif path_keys.length == 1 && path_keys.first != path_key
+        options[:environment][path_key] = options[:environment].delete(path_keys.first)
+      end
+
+      # Try our best to get rid of the "outer" omnibus ruby and any current
+      # chef-client/chef-dk installations from the path.
+      if ENV["MSYS_TOOLCHAIN"]
+        path_dirs = options[:environment].fetch(path_key, "").split(File::PATH_SEPARATOR || ":")
+        path_dirs = path_dirs.map { |p| windows_safe_path(p) }
+        safe_install_dir = windows_safe_path(install_dir)
+        path_dirs = path_dirs.reject do |p|
+          !p.start_with?(safe_install_dir) && filter_paths.any? { |f| p.start_with?(f) }
+        end
+        options[:environment][path_key] = path_dirs.join(File::PATH_SEPARATOR || ":")
+      end
```

Comment:
It might be nice to move the contents of this `if` block into a function that takes an environment and returns a new environment.